pkg/mcs/exportspec.go

```diff
@@ -180,7 +180,7 @@ func (es *ExportSpec) IsPreferredOver(another *ExportSpec) bool {
 // Retruns true when the specifications are compatible, otherwise returns
 // false and the name of the conflicting field.
 // @todo do we want to collect a map of the conflicting Global Properties?
-func (es *ExportSpec) IsCompatibleWith(another *ExportSpec) *CompatibilityError {
+func (es *ExportSpec) IsCompatibleWith(another *ExportSpec) error {
```

Comment:
Whats the reason for this change? if it can return an error other than CompatibilityError, it may be unrelated error that should be handled by the caller..

Reply:
It can return only CompatibilityError,
But returning the CompatibilityError type can cause a missing nil error (since error of CompatibilityError wont be recognize as nil).
Donald and Etai comment about it above and I changed it to return an error instead of CompatibilityError

Reply:
I cannot find the other comments on this.. and not sure I understand what is the problem? `*CompatibilityError` can also be nil right?  can you explain?

Reply:
I tagged you there,
It's a strange behavior, but if I understood it right the "nil" of `*CompatibilityError` isn't really nil (its a struct with the error field nil, so it wont get caught in a nil comparison)
https://go.dev/doc/faq#nil_error

Reply:
since this is VERY not intuitive, I strongly suggest returning a tuple - (bool CompatibilityError) instead of what is implemented now. This will also adhere to the convention that IsSomething() shall return a bool .. 
And I think that in this CompatibilityError should not even implement error interface. We can call if something like `ConflictDetails`.
@elevran @donaldh WDYT?

Reply:
`bool` and a conflict message (of type `string`) seem reasonable to me.